packages/db/index.js

```diff
@@ -130,26 +130,51 @@ export class DBClient {
     return data.length ? data[0] : undefined
   }
 
+  /**
+   * Check that a user account is restricted.
+   *
+   * @param {number} userId
+   * @returns {Promise<boolean>}
+   */
+  async isAccountRestricted (userId) {
+    return await this.getUserTagValue(userId, 'HasAccountRestriction') === 'true'
+  }
```

Comment:
```suggestion
  isAccountRestricted (userId) {
    return this.getUserTagValue(userId, 'HasAccountRestriction') === 'true'
  }
```

Given we not wrap this with try catch, no need to create extra promise layer with async await. We can just return the promise and ease a bit async callstack

Reply:
I tried this but then it complains

> This condition will always return 'false' since the types 'Promise<string>' and '"true"' have no overlap. Did you forget to use 'await'?

I'm not sure how we can avoid this since we're comparing it to `true` and we need the promise to be resolved before that. Let me know if you had something different in mind though.

For context: the user tag values are always a string and we don't know what type to expect. To avoid repeating logic, I created `getUserTagValue` to then use wrappers such as this to return the right type (in this case boolean, hence `=== 'true'`).